common/environment.go

```diff
@@ -60,3 +68,34 @@ func WriteEnvFile(env types.Environment, uidRange *user.UidRange, envFilePath st
 
 	return nil
 }
+
+// ReadEnvFileAsComposed reads the environment file, returning it as a
+// slice of strings, each expected but not checked to be of the form
+// "key=value".
+func ReadEnvFileAsComposed(envFilePath string) ([]string, error) {
+	var env []string
+	var envFile *os.File
+	var err error
+	if envFile, err = os.Open(envFilePath); err != nil {
+		return nil, err
+	}
+	defer envFile.Close()
+	scanner := bufio.NewScanner(envFile)
+	for scanner.Scan() {
+		env = append(env, scanner.Text())
+	}
+	if err = scanner.Err(); err != nil {
+		return nil, err
+	}
+	return env, nil
+}
+
+// ComposeEnviron formats the environment into a slice of strings,
+// each of the form "key=value".
+func ComposeEnviron(env types.Environment) []string {
```

Comment:
I would move part of the `defaultEnv` logic from `WriteEnvFile()` to here, so that it knows about default keys like `PATH` if missing from user configuration.